components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/recommendationmgt/AccessTokenGeneratorImpl.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.impl.recommendationmgt;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONObject;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AccessTokenGeneratorImpl implements AccessTokenGenerator {
+
+    private static final Log log = LogFactory.getLog(AccessTokenGeneratorImpl.class);
+
+    private static volatile AccessTokenGeneratorImpl accessTokenGenerator = null;
+    long generatedTime = 0;
+    long validityPeriod = 3600000;
+    String accessToken = null;
+    String oauthUrl = null;
+    String consumerKey = null;
+    String consumerSecret = null;
+
+    public AccessTokenGeneratorImpl() {
+
+    }
+
+    public static AccessTokenGeneratorImpl getInstance() {
+
+        if (accessTokenGenerator == null) {
+            synchronized (AccessTokenGeneratorImpl.class) {
+                if (accessTokenGenerator == null) {
+                    accessTokenGenerator = new AccessTokenGeneratorImpl();
+                }
+            }
+        }
+        return accessTokenGenerator;
+    }
+
+    @Override
+    public String getAccessToken(String oauthUrl, String consumerKey, String consumerSecret) {
+
+        if (this.oauthUrl == oauthUrl && this.consumerKey == consumerKey && this.consumerSecret == consumerSecret
```

Comment:
please use https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#equals(java.lang.Object) instead of ==

Reply:
Fixed